stac_fastapi/sqlalchemy/stac_fastapi/sqlalchemy/core.py

```diff
@@ -110,6 +110,12 @@ def item_collection(
         """Read an item collection from the database."""
         base_url = str(kwargs["request"].base_url)
         with self.session.reader.context_session() as session:
+            # Look up the collection first to get a 404 if it doesn't exist
+            collection = self._lookup_id(collection_id, self.collection_table, session)
+            if collection is None:
+                return NotFoundError(
+                    f"collection with id={collection_id} does not exist"
+                )
```

Comment:
This will never get hit. A `None` check and error raise is done `self._lookup_id()`, so just the lookup will work.

https://github.com/stac-utils/stac-fastapi/blob/e07aabb1169d2bbaf1215835405c33a6cfab039d/stac_fastapi/sqlalchemy/stac_fastapi/sqlalchemy/core.py#L59-L60

Reply:
Hm, you sure? I found that it returned `None` during my testing -- but let me take another look.

Reply:
Nope, you were right, I must have not tested things correctly. Thanks, good catch.